application/services/RawSubmission.php

```diff
@@ -11,6 +11,11 @@ public function addRawSubmission($details)
                 $details['headers'] = $headers;
             }
         }
+        for ($i=0;$i< sizeof($details['body']['mtbDetected']);$i++){
+            if($details['body']['mtbDetected'][$i]=='trace'){
+                $details['body']['rifResistance'][$i]='indeterminate';
+            }
+        }
```

Comment:
This doesn't belong here.

The purpose of this function was to save the values, in their raw format, that users submit to the server so that we can audit calculated results against values that were actually filled out by each user.

Please rather update the rifResistance field in Application_Service_Shipments inside updateTbResult & updateTbResults methods